### PR TITLE
minor reorganization, docker-compose for SQL

### PR DESCRIPTION
**Pull request by krishchow**
## [DEV-007] Dockerize Backend

Partial fix for: #7

-   [Y] Has the code been formatted?
-   [Y] Have dependencies/packages been added?

### Changes
This PR aims to dockerize SQL for use in the backend. The next step will be adding Redis. I also squashed some changes to the API Dockerfile, to better use caching provided by Docker and reduce build times.

### Testing
-   [X] Does the code have sufficient automated test coverage? If not, include reasoning and steps required to manually QA
 N/A, manually tested PSQL

### CODE REVIEW

File: storage/README.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
+# Pizza Storage
```
Comment (jcserv) [inline storage/README.md]: minor, change it to Dialog

**Review by jcserv — APPROVED** [review]
looks good!

File: storage/README.md (hunk 1 above)
Comment (krishchow) [reply storage/README.md]: fixed!

**Review by AipioxTechson — APPROVED** [review]
Looks good! There appears to be a merge conflict though

**Pull request merged**